api/types/src/response.rs

```diff
@@ -23,6 +26,18 @@ impl Response {
             body: serde_json::to_vec(body)?,
         })
     }
+
+    pub fn new_bcs<T: Serialize>(ledger_info: LedgerInfo, body: &T) -> Result<Self, Error> {
```

Comment:
You also need to change the content type below here to be different based on which type it is.

Reply:
My first try was to use a query string. But that's not clean. Http protocol uses "Accept" for a client to indicate what data format it can parse.